rfd/0217-bot-details-web.md

```diff
@@ -0,0 +1,330 @@
+
+As a member of the **Incident Response team**,
+I would like to **lock a bot** and all active instances,
+So that current and future access to protected resources is immediately prevented.
+
+As a member of the **Infrastructure Security team**,
+I would like to **view assigned roles and allowed traits** for a bot,
+So that I can easily determine the scope of access at a glance.
+
+As a member of the **Infrastructure Security team**,
+I would like to **view configured join methods** for a bot,
+So that I can easily determine the enrolment mechanisms available at a glance.
+
+As a member of the **Infrastructure Security team**,
+I would like to see a list of **currently active** instances for a bot with **last active times** and high-level characteristics (such as **bot name**, **hostname** and **join method**),
+So that I can assess the usage of the bot by scanning over the results to build an overall picture of recency and a distribution of characteristics.
+
+As a member of the **DevOps/Development team**, who has just set up an MWI agent (e.g. GitHub Actions),
+I would like to get **realtime feedback** that my new **agent/s are enrolling correctly**,
+So that I can feel confident that the agents will work going forwards.
+
+### UX
+
+![](assets/0217-iteration-1.png)
+
+#### Bot info and config
+
+Shows basic details and configuration. All items are readonly. Date/time items have a hover state which shows a tooltip with the full date and time.
+
+![](assets/0217-feature-info.png)
+
+**Data source**
+
+``` yaml
+# Bot resource (calculated)
+kind: bot
+metadata:
+  name: robot
+spec:
+  max_session_ttl:
+    seconds: 43200
+status:
+  role_name: bot-robot # Links to role
+  user_name: bot-robot # Links to user
+
+# User resource
+kind: user
+metadata:
+  labels:
+    teleport.internal/bot: robot # Links to bot
+  name: bot-robot
+spec:
+  created_by:
+    time: "2025-06-02T11:25:13.238653583Z"
+    user:
+      name: nicholas.marais@goteleport.com
+  roles:
+  - bot-robot # Links to role
+
+# Role resource
+kind: role
+metadata:
+  name: bot-robot
+```
+
+#### Join tokens
+
+Lists Join Tokens with a role of "Bot" and `bot_name` matching the bot being viewed. An overflow menu allows navigating to **Zero Trust Access > Join Tokens**. Clicking an item navigates to the view/edit page for that token.
+
+![](assets/0217-feature-join-tokens.png)
+
+To support MFA for Admin Actions, which includes viewing Join Tokens, a call to the endpoint will be made without the standard auto-challenge mechanism. If the call succeeds, the items are displayed without a prompt. If a requires-MFA error is returned, the UI will instead show an alert explaining that verification is required and provide an action to retry. On retry the usual auto-challenge mechanism kicks in to prompt the user, and the items are returned and displayed. This approach provides an improved user experience by giving the user full context before an MFA prompt appears.
```

Comment:
👍  This feels like a huge improvement to UX that we can probably ship to other pages eventually as well.